ako-infra/webhook/vks_cluster_webhook.go

```diff
@@ -0,0 +1,784 @@
+		VKSManagedLabel, VKSManagedLabelValueTrue,
+		cluster.GetNamespace(), cluster.GetName())
+
+	return &admissionv1.AdmissionResponse{
+		Allowed:   true,
+		Patch:     patchBytes,
+		PatchType: &patchType,
+		Result:    &metav1.Status{Message: "added VKS managed label"},
+	}
+}
+
+// shouldManageCluster determines if a cluster should be managed by VKS
+func (w *VKSClusterWebhook) shouldManageCluster(cluster *unstructured.Unstructured) bool {
+	// Check for explicit opt-out
+	labels := cluster.GetLabels()
+	if labels != nil {
+		if value, exists := labels[VKSManagedLabel]; exists && value == VKSManagedLabelValueFalse {
+			utils.AviLog.Infof("VKS webhook: cluster %s/%s opted out with label %s=%s",
+				cluster.GetNamespace(), cluster.GetName(), VKSManagedLabel, value)
+			return false
+		}
+	}
+
+	// Check if namespace has service engine group configuration
+	hasSEG, err := w.namespaceHasSEG(cluster.GetNamespace())
+	if err != nil {
+		utils.AviLog.Errorf("VKS webhook: failed to check SEG configuration for namespace %s: %v",
+			cluster.GetNamespace(), err)
+		return false
+	}
+
+	if !hasSEG {
+		utils.AviLog.Debugf("VKS webhook: namespace %s does not have SEG configuration", cluster.GetNamespace())
+		return false
+	}
+
+	utils.AviLog.Infof("VKS webhook: cluster %s/%s eligible for VKS management",
+		cluster.GetNamespace(), cluster.GetName())
+	return true
+}
+
+// namespaceHasSEG checks if a namespace has service engine group configuration
+func (w *VKSClusterWebhook) namespaceHasSEG(namespaceName string) (bool, error) {
+	namespace, err := w.client.CoreV1().Namespaces().Get(context.TODO(), namespaceName, metav1.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to get namespace %s: %w", namespaceName, err)
+	}
+
+	if namespace.Annotations != nil {
+		if _, exists := namespace.Annotations[internalLib.WCPSEGroup]; exists {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// createVKSLabelPatch creates a JSON patch to add the VKS managed AKO Install label
+func (w *VKSClusterWebhook) createVKSLabelPatch(cluster *unstructured.Unstructured) ([]map[string]interface{}, error) {
+	var patches []map[string]interface{}
+
+	labels := cluster.GetLabels()
+
+	if labels == nil {
+		patches = append(patches, map[string]interface{}{
+			"op":   "add",
+			"path": "/metadata/labels",
+			"value": map[string]string{
+				VKSManagedLabel: VKSManagedLabelValueTrue,
+			},
+		})
+	} else if existingValue, exists := labels[VKSManagedLabel]; exists {
+		if existingValue == VKSManagedLabelValueTrue {
+			return patches, nil
+		} else {
+			patches = append(patches, map[string]interface{}{
+				"op":    "replace",
+				"path":  "/metadata/labels/" + escapeJSONPointer(VKSManagedLabel),
+				"value": VKSManagedLabelValueTrue,
+			})
+		}
+	} else {
+		patches = append(patches, map[string]interface{}{
+			"op":    "add",
+			"path":  "/metadata/labels/" + escapeJSONPointer(VKSManagedLabel),
+			"value": VKSManagedLabelValueTrue,
+		})
+	}
```

Comment:
type for value is map in line 298 but on line 316 it is string, is this fine provided the op is add in both case.